### PR TITLE
UIIN-2596 Inventory - Staff suppress facet - default settings (follow-up)

### PR DESCRIPTION
**Pull request by BogdanDenis**
## Description
Re-work this feature to not have to call `onChange` callback inside filters. Calling this callback after mounting and searching caused another search to be executed, which closed Instance records after redirect
Now default filter is applied right after mounting by doing a single `history.replace`

## Screenshots

https://github.com/folio-org/ui-inventory/assets/19309423/4787ee0f-4d9f-4580-9eb4-053b4090eff8

## Issues
[UIIN-2596](https://folio-org.atlassian.net/browse/UIIN-2596)

## Related PRs
https://github.com/folio-org/stripes-components/pull/2234

### CODE REVIEW

File: src/components/InstanceFilters/InstanceFilters.test.js [hunk 1]
```diff
@@ -113,161 +107,157 @@ describe('InstanceFilters', () => {
...
-      });
+    afterEach(() => {
+      global.Storage.prototype.setItem.mockReset();
     });
 
-    it('should not render the facet', async () => {
+    it('should set a flag that user selected some option', async () => {
       renderInstanceFilters();
       const staffSuppressFacet = screen.queryByRole('button', { name: 'Staff suppress filter list' });
+      fireEvent.click(staffSuppressFacet);
```
Comment (mariia-aloshyna) [inline src/components/InstanceFilters/InstanceFilters.test.js:257]: `userEvent` is already used in this file, let's be consistent

File: src/components/InstancesList/InstancesList.js [hunk 2]
```diff
@@ -204,8 +205,12 @@ class InstancesList extends React.Component {
     }
 
     this.processLastSearchTerms();
+
+    this.applyDefaultStaffSuppressFilter();
   }
 
+
+
```
Comment (mariia-aloshyna) [inline src/components/InstancesList/InstancesList.js:213]: Remove extra line, please

**Review by mariia-aloshyna — APPROVED** [review]

**Comment by Dmytro-Melnyshyn** [comment]
1. Records are requested automatically, without submission on `Holdings` and `Item` segments.

https://github.com/folio-org/ui-inventory/assets/77053927/9e77f8ba-7d7b-41bb-b671-3ac6f49bb179

2. All requests for `Holdings` and `Item` segments are done with `staffSuppress.false`, is this correct?

**Comment by Dmytro-Melnyshyn** [comment]
There is a request on mount with having only the `filters=staffSuppress.false`



https://github.com/folio-org/ui-inventory/assets/77053927/e9c289e1-3ffc-4d9a-8a8a-b50f9c202d99

**Comment by BogdanDenis** [comment]
Hey @Dmytro-Melnyshyn 
1. I can't reproduce this with the same steps as on video, could you check again please?
2.  Yes this filter needs to be applied in all Search segments
3 and 4 also don't reproduce for me. Please make sure that you have the latest changes from this branch. After merging the previous PR I deleted the old branch and this one is new

**Review by Dmytro-Melnyshyn — APPROVED** [review]

**Pull request merged**